entry.go

```diff
@@ -29,6 +29,10 @@ type Entry struct {
 
 	// Message passed to Debug, Info, Warn, Error, Fatal or Panic
 	Message string
+
+	// A receive-only notification channel for knowing when all hooks have
+	// finished firing.
+	HooksDone <-chan struct{}
```

Comment:
If it's only for testing, I'd suggest doing the instrumentation in tests and not adding it to the type itself.